hera_cal/redcal.py

```diff
@@ -147,6 +150,71 @@ def get_reds(antpos, pols=['xx'], pol_mode='1pol', ex_ants=[], bl_error_tol=1.0,
     return add_pol_reds(pos_reds, pols=pols, pol_mode=pol_mode, ex_ants=ex_ants)
 
 
+def filter_reds(reds, bls=None, ex_bls=None, ants=None, ex_ants=None, ubls=None, ex_ubls=None, pols=None, ex_pols=None):
+    '''
+    Filter redundancies to include/exclude the specified bls, antennas, unique bl groups and polarizations.
+    Assumes reds indices are Antpol objects.
+    Args:
+        reds: list of lists of redundant baseline tuples, e.g. (ind1,ind2,pol)
+        bls (optional): baselines as antenna pair tuples (i,j) to include in reds.
+        ex_bls (optional): baselines as antenna pair tuples (i,j) to exclude in reds.
+        ants (optional): antenna numbers (as int's) to include in reds.
+        ex_ants (optional): antenna numbers (as int's) to exclude in reds.
+        ubls (optional): baselines representing their redundant group to include in reds.
+        ex_ubls (optional): baselines representing their redundant group to exclude in reds.
+        pols (optional): polarizations to include in reds. e.g. 'xy' or 'yx'.
+        ex_pols (optional): polarizations to exclude in reds. e.g. 'xy' or 'yx'.
+    Return:
+        reds: list of lists of redundant baselines as antenna pair tuples.
+    '''
+    if pols is None: # if no pols are provided, deduce them from the reds
+        pols = set(gp[0][2] for gp in reds)
+    if ex_pols:
+        pols = set(p for p in pols if p not in ex_pols)
+    reds = [gp for gp in reds if gp[0][2] in pols]
+    def expand_bls(gp):
+        gp3 = [(g[0],g[1],p) for g in gp if len(g) == 2 for p in pols]
+        return gp3 + [g for g in gp if len(g) == 3]
+    antpols = set(sum([list(split_pol(p)) for p in pols],[]))
+    def expand_ants(gp):
+        gp2 = [(g,p) for g in gp if not hasattr(g,'__len__') for p in antpols]
+        return gp2 + [g for g in gp if hasattr(g,'__len__')]
+    def split_bls(bls):
+        return [split_bl(bl) for bl in bls]
+    if ubls or ex_ubls:
```

Comment:
I find this to be more difficult to parse than `if (ubls is not None) or (ex_ubls is not None):`. Even though lists and strings evaluate to `True`, it's a quirk of python that's not particularly intuitive.

Reply:
Also, same for `if ex_pols` above and more examples below.